com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.Types.cs

```diff
@@ -245,14 +245,17 @@ public HDLightType type
                         case HDLightType.Directional:
                             legacyLight.type = LightType.Directional;
                             m_PointlightHDType = PointLightHDType.Punctual;
+                            HDLightRenderDatabase.instance.GetLightData(lightEntity).pointLightType = m_PointlightHDType;
```

Comment:
More of a discussion point rather than a review comment per se, but should we maybe have the getter for a _modifiable_ pointer named differently? I wonder if we can find a way to have this more explicitly showing that we are actually modifying underlying data. 

I don't think how it is now is necessarily bad, but I wonder if we could make this more explicit in showing that it could lead to modifications too.

Reply:
I think even GetLightDataRef() could be a bit better? Though not a great name :D

Reply:
Replaced this with EditLightDataAsRef(). Added another one called GetLightDataAsRef().
Because c# cant have readonly refs, at the least put these methods for perf & convension.